File: Miscellaneous_Permutation.vb
Module Miscellaneous_Permutation

    Sub Main()

        Dim goods
        goods = New List(Of String)(New String() {"Eno", "Chidi", "Olu", "Ahmed", "Osas", "Gbeda"})
        Dim perm As New Permutation
        Dim result As List(Of String()) = perm.possibleWordPermutations(goods, 3)
        ' print choices and operation
        Console.Write("[ ")
        For Each choice As String In perm.words
            Console.Write(choice & "; ")
        Next
        Console.WriteLine("] permutation " & perm.r & ":" & Environment.NewLine)
        ' print out permutations nicely
        For Each group As String() In result
            For Each member As String In group
                Console.Write(member & "; ")
            Next
            Console.WriteLine()
        Next
        Console.WriteLine(Environment.NewLine & "Number of ways is " & result.Count & ".")

    End Sub

End Module

File: Miscellaneous_Permutation.cpp
#include "stdafx.h"
#include "Permutation.h"
#include <vector>

#include <iostream>

using namespace std;

int main()
{
	vector<string> goods = { "Eno", "Chidi", "Olu", "Ahmed", "Osas", "Gbeda" };
	Permutation perm;
	vector<vector<string>> result = perm.possibleWordPermutations(goods, 3);
	// print choices and operation
	cout << "\n[ ";
	for (string choice : perm.words) {
		cout << choice << "; ";
	}
	cout << "] permutation " << perm.r << ":\n\n";

	// print out permutations nicely
	for (vector<string> set : result) {
		for (string member : set) {
			cout << member << "; ";
		}
		cout << "\n";
	}
	cout << "\nNumber of ways is " << result.size() << ".";

    return 0;
}

